hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -96,6 +97,8 @@ public HoodieTableConfig(FileSystem fs, String metaPath, String payloadClassName
       throw new HoodieIOException("Could not load Hoodie properties from " + propertyPath, e);
     }
     this.props = props;
+    ValidationUtils.checkArgument(props.containsKey(HOODIE_TABLE_TYPE_PROP_NAME) && props.containsKey(HOODIE_TABLE_NAME_PROP_NAME),
```

Comment:
sorry, I don't get why we need this here. If properties contain table type and table name, why bail out?

Reply:
these properties are written always. so this indicates a corrupted file (with high probability) . Actually what we discussed is that there wont be a corrupted/partial hoodie.properties file given even s3 (or gcs) guarantees partial writes won't be visible. So this is just additional safety

Reply:
>sorry, I don't get why we need this here. If properties contain table type and table name, why bail out?

we bail out if these props are not present.

Reply:
ok, was confused as to how this related to this patch of upgrade/downgrade.